ledger/heavy/executor/finalization_keeper.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package executor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/pulse"
+	"github.com/insolar/insolar/instrumentation/inslogger"
+	"github.com/pkg/errors"
+)
+
+//go:generate minimock -i github.com/insolar/insolar/ledger/heavy/executor.FinalizationKeeper -o ./ -s _gen_mock.go
+
+// FinalizationKeeper check how far from each other last finalized pulse and current one
+// and if distance is more than limit it stops network
+type FinalizationKeeper interface {
+	OnPulse(ctx context.Context, current insolar.PulseNumber) error
+}
+
+type FinalizationKeeperDefault struct {
+	jetKeeper       JetKeeper
+	networkStopper  insolar.TerminationHandler
+	limit           int
+	pulseCalculator pulse.Calculator
+}
+
+func NewFinalizationKeeperDefault(jk JetKeeper, ns insolar.TerminationHandler, pc pulse.Calculator, limit int) *FinalizationKeeperDefault {
+	return &FinalizationKeeperDefault{
+		jetKeeper:       jk,
+		networkStopper:  ns,
+		limit:           limit - 1,
```

Comment:
`limit - 1` can be confusing, should we pass already calculated finalization limit outside c-tor?

Reply:
I move this subtraction to function code